packages/edit-post/src/components/header/template-title/delete-template.js

```diff
@@ -58,11 +53,10 @@ export default function DeleteTemplate() {
 			template: '',
 		} );
 		const settings = getEditorSettings();
-		const newAvailableTemplates = pickBy(
-			settings.availableTemplates,
-			( _title, id ) => {
-				return id !== template.slug;
-			}
+		const newAvailableTemplates = Object.fromEntries(
+			Object.entries( settings.availableTemplates ?? {} ).filter(
```

Comment:
`settings.availableTemplates` could potentially be nullish, therefore we're adding a fallback to keep the original `_.pickBy()` behavior of always returning an object.